tests/e2e/utils/restart.ts

```diff
@@ -1,19 +1,25 @@
-import { Page, expect } from '@playwright/test';
+import { type Page, test } from '@playwright/test';
 
 // Function to check modal visibility
 const isModalVisible = async (page: Page) => {
-  return page.isVisible('text="Waiting for restart..."');
+  return await Promise.any([
+    page.waitForSelector('text="Waiting for restart..."', { state: 'visible' }),
+    page.waitForSelector('text="is taking longer"', { state: 'visible' }),
+  ])
+    .then(() => true)
+    .catch(() => false);
 };
 
 /**
- * Wait for a restart modal to appear, but instead of failing if it doesn't, attempt to
- * refresh the page and see if it comes back up
+ * Wait for a restart modal to appear and then disappear
```

Comment:
The reload was originally introduced to reduce flakiness, but now it is causing tests to fail; it shouldn't have been needed to begin with so I've removed it